install_config/router/customized_haproxy_router.adoc

```diff
@@ -13,48 +13,666 @@ toc::[]
+xref:using-configmap-replace-template[*configMap*] as a volume in the router
+pod.  The *TEMPLATE_FILE* environment
+varuable is set to the full path name of the file in the router pod.
+
+An alternative approach is to modify the *haproxy-template.config* file
+and xref:rebuilding-your-router[rebuild] the router image. The new image
+is pushed to the the cluster repositiory and the routers deployment
+configuration *image:* field is updated with the new name. When the cluster is
+updated the image needs to be rebuilt and pushed.
+
+In either case it starts with the template file.
 
 [[obtaining-router-configuration-template]]
 == Obtaining the Router Configuration Template
 
-You can obtain a new *_haproxy-config.template_* file from the latest router
-image by running:
+The HAProxy template file is fairly large and complex. For some changes it may
+be easier to modify the existing template rather than writing a complete
+replacement.  You can obtain a *_haproxy-config.template_* file from a running
+router by logging onto the node that is running the router:
 
 ----
 ifdef::openshift-enterprise[]
 # docker run --rm --interactive=true --tty --entrypoint=cat \
-    registry.access.redhat.com/openshift3/ose-haproxy-router:v3.0.2.0 haproxy-config.template
+    registry.access.redhat.com/openshift3/ose-haproxy-router:v3.5.0.0 haproxy-config.template
 endif::[]
 ifdef::openshift-origin[]
 # docker run --rm --interactive=true --tty --entrypoint=cat \
     openshift/origin-haproxy-router haproxy-config.template
 endif::[]
 ----
 
+Where the image name is from "docker images"
+or by running this on master referencing the router pod:
+
+----
+# oc get po
+NAME                       READY     STATUS    RESTARTS   AGE
+router-2-40fc3             1/1       Running   0          11d
+# oc rsh router-2-40fc3 cat haproxy-config.template > haproxy-config.template
+# oc rsh router-2-40fc3 cat haproxy.config > haproxy.config
+----
+
 Save this content to a file for use as the basis of your customized template.
+The saved haproxy.config shows what is actually running.
+
+
+[[router-configuration-template]]
+== Modifying the Router Configuration Template
+
+[[router-template-background]]
+
+=== Background
+
+The template is based on the
+link:https://golang.org/pkg/text/template/[go template]. It can reference any
+of the environment variables in the router's deployment configuration, any
+config information that is described below, and haproxy router provided helper
+functions.
+
+The structure of the template file mirrors the resulting HAProxy config file.
+As the template is processed anything not surrounded by "{{" something "}}"
+is directly copied to the config file. Passages that are surrounded by "{{"
+something "}}" are evaluated and the resulting text, if any, is copied to the
+config file.
+
+[[go-template]]
+
+=== Go Template Constructs
+
+The following excerpts are from the
+link:https://golang.org/pkg/text/template/[go template]
```

Comment:
I did copy from the referenced doc, just giving credit and pointing to a more through discussion.